<?xml version="1.0" ?>
<robot name="Robot" xmlns:xacro="http://www.ros.org/wiki/xacro">

<xacro:include filename="$(find Robot_description)/urdf/materials.xacro" />
<xacro:include filename="$(find Robot_description)/urdf/Robot.trans" />
<xacro:include filename="$(find Robot_description)/urdf/Robot.gazebo" />
<link name="base_link">
  <inertial>
    <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
    <mass value="0.0"/>
    <inertia ixx="0.0" iyy="0.0" izz="0.0" ixy="0.0" iyz="0.0" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="ssplaca11111_1">
  <inertial>
    <origin xyz="0.16816144736085598 1.130242303244973 0.08435717350176891" rpy="0 0 0"/>
    <mass value="375.40867935896205"/>
    <inertia ixx="38.020673" iyy="20.791186" izz="56.844586" ixy="-8e-06" iyz="-0.118171" ixz="-0.0"/>
  </inertial>
  <visual>
    <origin xyz="-0.37818 1.900695 0.558373" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/ssplaca11111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.37818 1.900695 0.558373" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/ssplaca11111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="BASE111_1">
  <inertial>
    <origin xyz="-0.0819064610691344 -0.00043883275899025964 0.11638560527943387" rpy="0 0 0"/>
    <mass value="570.6700095274523"/>
    <inertia ixx="15.007732" iyy="29.800288" izz="35.945756" ixy="-0.008386" iyz="-0.049965" ixz="0.974627"/>
  </inertial>
  <visual>
    <origin xyz="-0.546341 0.764805 0.423668" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/BASE111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.546341 0.764805 0.423668" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/BASE111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="S_AXIS1111111111_1">
  <inertial>
    <origin xyz="0.24466794479935539 0.05128279784333545 0.20193877691570378" rpy="0 0 0"/>
    <mass value="704.6086442897418"/>
    <inertia ixx="65.135386" iyy="100.672217" izz="129.875461" ixy="-30.931057" iyz="8.201024" ixz="10.585126"/>
  </inertial>
  <visual>
    <origin xyz="-0.538792 0.764805 0.131668" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/S_AXIS1111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.538792 0.764805 0.131668" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/S_AXIS1111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="L_AXIS1111111111_1">
  <inertial>
    <origin xyz="0.2953257448252329 -0.05781067633528614 0.39152298403718966" rpy="0 0 0"/>
    <mass value="552.7941542901582"/>
    <inertia ixx="65.217806" iyy="82.204935" izz="25.704715" ixy="-3.072531" iyz="-3.911626" ixz="-30.518136"/>
  </inertial>
  <visual>
    <origin xyz="-1.256357 0.567865 -0.016332" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/L_AXIS1111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-1.256357 0.567865 -0.016332" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/L_AXIS1111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="U_AXIS1111111111_1">
  <inertial>
    <origin xyz="0.12105319321935815 0.2774925707551613 0.24827752694931782" rpy="0 0 0"/>
    <mass value="483.423381953783"/>
    <inertia ixx="35.212477" iyy="50.339318" izz="40.849864" ixy="-13.090506" iyz="-13.220022" ixz="-18.661963"/>
  </inertial>
  <visual>
    <origin xyz="-1.854654 0.536771 -1.015979" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/U_AXIS1111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-1.854654 0.536771 -1.015979" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/U_AXIS1111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="R_AXIS1111111111_1">
  <inertial>
    <origin xyz="0.1808151825540092 -0.0015520884227349686 0.03478524471465416" rpy="0 0 0"/>
    <mass value="8.899939703677402"/>
    <inertia ixx="0.051132" iyy="0.090197" izz="0.08962" ixy="0.00492" iyz="0.001768" ixz="-0.004746"/>
  </inertial>
  <visual>
    <origin xyz="-2.556688 -0.02455 -1.635389" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/R_AXIS1111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-2.556688 -0.02455 -1.635389" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/R_AXIS1111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="B_AXIS1111111111_1">
  <inertial>
    <origin xyz="-0.0007938014575432639 0.10047052116865882 0.13744793458112237" rpy="0 0 0"/>
    <mass value="45.97892583556151"/>
    <inertia ixx="0.48222" iyy="0.52207" izz="0.447959" ixy="0.04733" iyz="-0.009258" ixz="-0.080339"/>
  </inertial>
  <visual>
    <origin xyz="-2.80176 0.03456 -1.648816" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/B_AXIS1111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-2.80176 0.03456 -1.648816" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/B_AXIS1111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="T_AXIS111111111_1">
  <inertial>
    <origin xyz="0.008292396655664724 -0.0005762393489285034 0.011520103720092711" rpy="0 0 0"/>
    <mass value="5.836276011069453"/>
    <inertia ixx="0.017241" iyy="0.013058" izz="0.0211" ixy="-0.000287" iyz="-0.000404" ixz="0.005763"/>
  </inertial>
  <visual>
    <origin xyz="-2.869373 -0.0574 -1.876035" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/T_AXIS111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-2.869373 -0.0574 -1.876035" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/T_AXIS111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="BALANCER_11111111111_1">
  <inertial>
    <origin xyz="0.22642099632217927 0.10875488726730087 -0.20303725456543642" rpy="0 0 0"/>
    <mass value="1052.0941458276736"/>
    <inertia ixx="39.017297" iyy="119.796715" izz="128.357793" ixy="-37.307883" iyz="9.488315" ixz="24.255809"/>
  </inertial>
  <visual>
    <origin xyz="-0.268876 1.096595 -0.599527" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/BALANCER_11111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.268876 1.096595 -0.599527" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/BALANCER_11111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="BALANCER_21111111111_1">
  <inertial>
    <origin xyz="-0.19378739451528237 -1.09478338404662 0.32649982944380457" rpy="0 0 0"/>
    <mass value="34.72062884447938"/>
    <inertia ixx="1.166404" iyy="5.632623" izz="6.085232" ixy="-2.053086" iyz="0.520178" ixz="1.334368"/>
  </inertial>
  <visual>
    <origin xyz="-0.8843 -0.182913 -0.020402" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/BALANCER_21111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.8843 -0.182913 -0.020402" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://Robot_description/meshes/BALANCER_21111111111_1.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<joint name="base_link1" type="prismatic">
  <origin xyz="0.37818 -1.900695 -0.558373" rpy="0 0 0"/>
  <parent link="base_link"/>
  <child link="ssplaca11111_1"/>
  <axis xyz="0.0 -1.0 0.0"/>
  <limit upper="1.3" lower="-0.4" effort="100" velocity="100"/>
</joint>

<joint name="Rígida 23" type="fixed">
  <origin xyz="0.168161 1.13589 0.134705" rpy="0 0 0"/>
  <parent link="ssplaca11111_1"/>
  <child link="BASE111_1"/>
</joint>

<joint name="link1_link2" type="continuous">
  <origin xyz="-0.007549 0.0 0.292" rpy="0 0 0"/>
  <parent link="BASE111_1"/>
  <child link="S_AXIS1111111111_1"/>
  <axis xyz="0.0 -0.0 1.0"/>
</joint>

<joint name="link2_link3" type="revolute">
  <origin xyz="0.717565 0.19694 0.148" rpy="0 0 0"/>
  <parent link="S_AXIS1111111111_1"/>
  <child link="L_AXIS1111111111_1"/>
  <axis xyz="0.364297 -0.931283 -0.0"/>
  <limit upper="0.698132" lower="-1.308997" effort="100" velocity="100"/>
</joint>

<joint name="link3_link4" type="revolute">
  <origin xyz="0.598297 0.031094 0.999647" rpy="0 0 0"/>
  <parent link="L_AXIS1111111111_1"/>
  <child link="U_AXIS1111111111_1"/>
  <axis xyz="-0.364297 0.931283 -0.0"/>
  <limit upper="1.919862" lower="-1.396263" effort="100" velocity="100"/>
</joint>

<joint name="link4_link5" type="continuous">
  <origin xyz="0.702034 0.561321 0.61941" rpy="0 0 0"/>
  <parent link="U_AXIS1111111111_1"/>
  <child link="R_AXIS1111111111_1"/>
  <axis xyz="0.862521 0.337398 0.377121"/>
</joint>

<joint name="link5_link6" type="revolute">
  <origin xyz="0.245072 -0.05911 0.013427" rpy="0 0 0"/>
  <parent link="R_AXIS1111111111_1"/>
  <child link="B_AXIS1111111111_1"/>
  <axis xyz="-0.476887 0.791219 0.382819"/>
  <limit upper="2.181662" lower="-2.181662" effort="100" velocity="100"/>
</joint>

<joint name="link6_link7" type="continuous">
  <origin xyz="0.067613 0.09196 0.227219" rpy="0 0 0"/>
  <parent link="B_AXIS1111111111_1"/>
  <child link="T_AXIS111111111_1"/>
  <axis xyz="0.583709 -0.04055 0.81095"/>
</joint>

<joint name="Rígida 31" type="fixed">
  <origin xyz="-0.615424 -1.279508 0.579125" rpy="0 0 0"/>
  <parent link="BALANCER_21111111111_1"/>
  <child link="BALANCER_11111111111_1"/>
</joint>

<joint name="Rígida 32" type="fixed">
  <origin xyz="0.345508 0.947718 0.15207" rpy="0 0 0"/>
  <parent link="S_AXIS1111111111_1"/>
  <child link="BALANCER_21111111111_1"/>
</joint>

</robot>
